bundles/framework/myplaces3/view/MainView.js

```diff
@@ -107,8 +107,10 @@ Oskari.clazz.define('Oskari.mapframework.bundle.myplaces3.view.MainView',
                 if (!event.getSticky()) {
                     return;
                 }
-                // changed tool
+
+                // changed tool -> clean popup
                 this.cleanupPopup();
+                this.form.destroy();
```

Comment:
Should form.destroy() be in cleanPopup()?

Reply:
Moved to cleanPopup().